Add connector config documentation (#100)
wso2-extensions/identity-inbound-auth-sts: docs/config.md

# Configuring WS-Trust Security Token Service
WSO2 Identity Server uses the security token service (STS) as the WS-Trust implementation. The STS is capable of issuing
SAML 1.1 and 2.0 security tokens and has a SOAP/XML API for token issuance. This section provides the instructions to 
configure WS-Trust Security Token Service with WSO2 Identity Server. 

> ## Installing the Connector

Reviewer: ```suggestion
## Installing the connector
```

   1. Download the connector from [WSO2 Connector Store](https://store.wso2.com/store/assets/isconnector/list).

>    2. Navigate to the PRODUCT_HOME and paste the `wso2is-sts-connector-x.x.x.zip` file downloaded from the WSO2 

Reviewer: ```suggestion
   2. Navigate to the `<PRODUCT_HOME>`, paste the `wso2is-sts-connector-x.x.x.zip` file downloaded from the WSO2 
```

> Connector Store and extract it.

Reviewer: ```suggestion
Connector Store and extract it. The extracted folder will be referred to as `<CONNECTOR_HOME>` in the rest of this document. 
```

>    3. If you are using MacOS/Ubuntu, navigate to the extracted folder and execute the following commands.

Reviewer: ```suggestion
   3. If you are using MacOS/Ubuntu, navigate to `<CONNECTOR_HOME>` and execute the following commands.
```

            
            chmod u+r+x setup_sts.sh
            ./setup_sts.sh 
            
      Else,

>         - Navigate to the `extracted_folder/dropins` and copy the jars in that location to 

Reviewer: ```suggestion
        - Navigate to the `<CONNECTOR_HOME>/dropins` and copy the jars in that location to 
```

>         `PRODUCT_HOME/repository/components/dropins`.

Reviewer: ```suggestion
        `<PRODUCT_HOME>/repository/components/dropins`.
```

>         - Navigate to the `extracted_folder/deployment-client-modules` and copy the .mar file in that location to 

Reviewer: ```suggestion
        - Navigate to the `<CONNECTOR_HOME>/deployment-client-modules` and copy the .mar file in that location to 
```

>         `PRODUCT_HOME/repository/deployment/client/modules`.

Reviewer: ```suggestion
        `<PRODUCT_HOME>/repository/deployment/client/modules`.
```

>    4. Navigate back to the PRODUCT_HOME and delete the extracted_folder. Now you have successfully installed the 

Reviewer: ```suggestion
   4. Navigate back to `<PRODUCT_HOME>` and delete `<CONNECTOR_HOME` folder . Now you have successfully installed the 
```

   connector.
   

> ## Configuration Steps

Reviewer: ```suggestion
## Configuration steps
```

   to your requirements. For this tutorial, use `UsernameToken` under the **Basic Scenarios** section.

   5. Click **Next**. The user domain and user group selection appears.

   6. Provide the required details as follows:
        1. Select **ALL-USER-STORE-DOMAINS**.
        2. Select the role you created to grant permission to access secured service. In this example, the `admin` role
         is used. Next, click **Finish**.
        
   7. Click **Finish**.

   8. Click **Ok** on the confirmation dialog window that appears.

   9. Click **Update** to complete the process.
   
Now STS is configured and secured with a username and password. Only users with the `admin` role can consume the 
service.

The next step is to add a service provider to consume the STS.

#### Adding a service provider for the STS client

**Before you begin:** You must first
    [register a service provider](https://is.docs.wso2.com/en/5.11.0/learn/adding-and-configuring-a-service-provider/#adding-a-service-provider).
    
    To register a service provider:
	    1. Sign in to WSO2 Identity Server Management Console as an admin.

> 	    2. On the Main menu, click Identity > Service Providers > Add.

Reviewer: ```suggestion
	    2. On the Main menu, click **Identity > Service Providers > Add**.
```

	    3. Enter a service provider name.

> 	    4. Click Register. The Service Provider Details page appears.

Reviewer: ```suggestion
	    4. Click **Register**. The Service Provider Details page appears.
```

 	
        <table>
           <thead>
              <tr class="header">
                 <th>Field</th>
                 <th>Description</th>
                 <th>Sample Value</th>
              </tr>
           </thead>
           <tbody>
              <tr class="odd">
                 <td><strong>Endpoint Address</strong></td>
                 <td>
                    <div class="content-wrapper">
                       <div>
                          Enter the trusted relying party's <strong>endpoint address,</strong> which is the <strong></strong> endpoint address of the Security Token Service.
                          <p>The endpoint must be used as the service <code>                  URL                 </code> to which the token gets delivered by the STS client. Then select the public certificate imported. Tokens issued are encrypted using the public certificate of the trusted relying party. Therefore, the consumer who obtains this token, to invoke the RP service, will not be able to see the token.</p> 
                           <div class="admonition note">
                            <p class="admonition-title"><strong>Note</strong></p>

>                                  Make sure to upload the certificate of the relying party to the truststore. For instructions, see <a href="https://docs.wso2.com/display/ADMIN44x/Creating+New+Keystores#CreatingNewKeystores-ca_certificateAddingCA-signedcertificatestokeystores">Adding CA-signed certificates to keystores</a>.</div>

Reviewer: ```suggestion
                                 Make sure to upload the certificate of the relying party to the truststore. For instructions, see <a href="https://is.docs.wso2.com/en/latest/administer/creating-new-keystores/#adding-ca-signed-certificates-to-keystores">Adding CA-signed certificates to keystores</a>.</div>
```

                          <br/>
                       </div>
                    </div>
                 </td>
                 <td><code>               https://localhost:9444/services/echo              </code></td>
              </tr>
              <tr class="even">
                 <td><strong>Certificate Alias</strong></td>
                 <td>This is the alias of the certificate.</td>
                 <td><code>               wso2carbon              </code></td>
              </tr>
           </tbody>
        </table>

   3. Click **Update** to save the changes made to the service provider.
   
#### Running the STS client
The following sample demonstrates the steps required to run a Security Token Service (STS) client. The STS provides 
the service of issuing a security token to authenticate a user for your client application, using WS-Trust protocols.

   1. The code for the sample can be checked out from the [GitHub repository](https://github.com/wso2/samples-is).

>    2. Once downloaded the identity server samples, go inside to `SAMPLES-IS/sts/sts-client` directory to build the STS 

Reviewer: ```suggestion
   2. Once downloaded the identity server samples, go inside the `<SAMPLES-IS>/sts/sts-client` directory to build the STS 
```
